### ISSUE TITLE
Research network request optimization using Orbit.js

### ISSUE THREAD

**Issue by jugglinmike** [bocoup/nest-weekly-review#19]
[The Orbit.js library](https://github.com/orbitjs/orbit.js) should make it possible for clients to avoid requesting redundant information. This may require [JSON Patch](http://jsonpatch.com/) support in Bocoup's API; I just don't know.

**Comment by jugglinmike** [comment]
I started down this path, and while I think it could be really cool, I've hit some difficulties:

- I immediately encountered a bug in the LocalStorage "source" which halted development for a few hours. This issue was reported over two months ago and has yet to be acknowledged by the project maintainers: https://github.com/orbitjs/orbit.js/issues/79.
- Another bug appears in an internal method which [has been labeled "TODO - test" for six months](https://github.com/orbitjs/orbit.js/blob/4332c57a6e1c83642413f2d8f0758d88607c27e8/lib/orbit-common/schema.js#L409-L433).
- The project's "readme" file is sorely out of date (see https://github.com/orbitjs/orbit.js/pull/89), reflecting a larger documentation problem that may continue to deter integration efforts.

I don't think any of these are show-stoppers, and I think that Orbit has a lot of potential. Given that this represents non-trivial and low-level application restructuring, I'm having second thoughts about to the library today.

In light of this, I decided to address the most pressing concern (conformance with the JSON API working draft) manually (see gh-56). In addition to the benefits listed in the issue description, Orbit.js's support for the JSON API would  smooth out integrating with the Nest API in the future. The weight of that advantage isn't obvious, though; it may be that the draft, the Nest API server, *and* this application are now "stable enough".
